### PR TITLE
Bump serde to 1.0

### PR DESCRIPTION
**Pull request by SimonSapin**
This is a rebase of #17325 with `[replace]` entries removed, a bunch more dependencies updated, and some more compile fixes. Original work by @Eijebong, thanks a lot!

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17355)

### CODE REVIEW

File: components/constellation/constellation.rs [hunk 1]
```diff
@@ -479,7 +479,7 @@ const WARNINGS_BUFFER_SIZE: usize = 32;
 /// but does not panic on deserializtion errors.
 fn route_ipc_receiver_to_new_mpsc_receiver_preserving_errors<T>(ipc_receiver: IpcReceiver<T>)
     -> Receiver<Result<T, IpcError>>
-    where T: Deserialize + Serialize + Send + 'static
+    where T: for<'de> Deserialize<'de> + Serialize + Send + 'static
```
Comment (nox) [inline components/constellation/constellation.rs]: @dtolnay Shouldn't this be `DeserializeOwned`?

File: components/script/script_thread.rs [hunk 2]
```diff
@@ -2322,7 +2322,7 @@ impl ScriptThread {
     }
 
     fn handle_css_error_reporting(&self, pipeline_id: PipelineId, filename: String,
-                                  line: usize, column: usize, msg: String) {
+                                  line: u32, column: u32, msg: String) {
```
Comment (nox) [inline components/script/script_thread.rs]: How is this change related?

Comment (SimonSapin) [inline components/script/script_thread.rs]: It’s not. https://github.com/servo/rust-cssparser/pull/157 landed in cssparser in the mean time and Servo was not updated for it.
